src/contract.ts

```diff
@@ -54,10 +52,7 @@ export async function getContractABI(name: string, build_dir: string, trace: boo
 export async function getContract(web3: Web3, name: string, build_dir: string, trace: boolean, defaultOptions: SendOptions): Promise<Contract> {
   const contractBuild = await getContractBuild(name, build_dir, trace);
   const contractAbi = JSON.parse(contractBuild.abi);
-  (<any>newContract).setProvider((<any>web3).currentProvider);
-  const contract = <Contract>(new (<any>newContract)(contractAbi));
-
-  return contract;
+  return new web3.eth.Contract(contractAbi, undefined, defaultOptions);;
```

Comment:
Hm, any idea why this was done the old way (minus the type issues)?

Reply:
I have a feeling that it felt better to use the subdependancy directly, but unfortunately the name setProvider for the subdependancy is non obvious that you need to also pass in the unlocked accounts https://github.com/ethereum/web3.js/blob/1.x/packages/web3-eth-contract/src/index.js#L301